src/SourceBuild/content/eng/tools/CreateBaselineUpdatePR/PRCreator.cs

```diff
@@ -280,72 +280,120 @@ private async Task<TreeResponse> CreateParentTreeAsync(TreeResponse testResultsT
+            newBranchName = existingPullRequest.Head.Ref;
 
-            // Update the body of the pull request
-            var pullRequestUpdate = new PullRequestUpdate
+            try
             {
-                Body = pullRequestBody
-            };
-            await _client.PullRequest.Update(_repoOwner, _repoName, matchingPullRequest.Number, pullRequestUpdate);
+                // Merge the target branch into the existing pull request
+                var merge = new NewMerge(newBranchName, headSha);
+                var mergeResult = await _client.Repository.Merging.Create(_repoOwner, _repoName, merge);
+            }
+            catch (Exception e)
+            {
+                Log.LogWarning($"Failed to merge the target branch into the existing pull request: {e.Message}");
+                Log.LogWarning("Continuing with updating the existing pull request. You may need to resolve conflicts manually in the PR.");
+            }
 
-            Log.LogInformation($"Updated existing pull request #{matchingPullRequest.Number}. URL: {matchingPullRequest.HtmlUrl}");
+            headSha = await GetHeadShaAsync(newBranchName);
         }
-        else
+
+        var commitSha = await CreateCommitAsync(parentTreeResponse.Sha, headSha, $"Update baselines for build {BuildLink}{buildId} (internal Microsoft link)");
+        if (await ShouldMakeUpdatesAsync(headSha, commitSha))
         {
-            var comparison = await _client.Repository.Commit.Compare(_repoOwner, _repoName, headReference.Object.Sha, commitResponse.Sha);
-            if (!comparison.Files.Any())
+            string pullRequestBody = $"This PR was created by the `CreateBaselineUpdatePR` tool for build {buildId}. \n\n" +
+                                 $"The updated test results can be found at {BuildLink}{buildId} (internal Microsoft link)";
+            if (existingPullRequest != null)
             {
-                // No changes to commit
-                Log.LogInformation("No changes to commit. Skipping PR creation.");
-                return;
+                await UpdatePullRequestAsync(newBranchName, commitSha, pullRequestBody, existingPullRequest);
             }
+            else
+            {
+                await CreatePullRequestAsync(newBranchName, commitSha, targetBranch, title, pullRequestBody);
+            }
+        }
+    }
 
-            // Create a new pull request
-            var newReference = new NewReference("refs/heads/" + newBranchName, commitResponse.Sha);
-            await _client.Git.Reference.Create(_repoOwner, _repoName, newReference);
+    private async Task<PullRequest?> GetExistingPullRequestAsync(string title, string targetBranch)
+    {
+        var request = new PullRequestRequest
+        {
+            Base = targetBranch
+        };
+        var existingPullRequest = await _client.PullRequest.GetAllForRepository(_repoOwner, _repoName, request);
+        return existingPullRequest.FirstOrDefault(pr => pr.Title == title);
```

Comment:
I see that the PR titles being passed in do not include a target branch name in the title. This pattern is commonly done for things like dependency flow PRs. Consider doing that. Otherwise, this logic needs to be refined so that it can handle PR title collisions when multiple PRs exist for different branches.

Reply:
The target branch is included in the `PullRequestRequest`:

```C#
  var request = new PullRequestRequest
  {
      Base = targetBranch
  };
   var existingPullRequest = await _client.PullRequest.GetAllForRepository(_repoOwner, _repoName, request);
```

Since the request is passed into the call to get the associated PRs, it should only consider PRs that are based on the target branch.